📜Create tutorials (#4)
KinsonDigital/VelaptorDocs: docs/tutorials/intro.mdx

sidebar_position: 1
---

# Intro to Velaptor

Velaptor is an open-source framework for building high-performant, high-quality, cross-platform 2D games using C#/dotnet.

:::info
[Here](https://www.freecodecamp.org/news/the-difference-between-a-framework-and-a-library-bd133054023f/) is a good explanation of a framework vs a library.
:::

## Overview

> Velaptor focuses on developing your game.  What this means is that instead of dealing with low-level details such as writing low level
> code for rendering graphics, keyboard and mouse input, sound, and other common things required by games, you can just focus on writing your game.
> 
> Writing these things from scratch can take hours when all you want to do is create.  This is where Velaptor can help!!

Reviewer: Grammar
```suggestion
Velaptor focuses on creating your game instead of building time-consuming details.  Writing low level code for rendering graphics, keyboard and mouse input, sound, and other common things required by games can take hours (trust me, I know).  With Velaptor, you can just focus on writing your game!!
```

Author: I changed it to the suggestion below.  Let me know what you think.

```suggestion
Velaptor focuses on creating your game instead of having to handle low-level and time-consuming details.  Writing low-level code for rendering graphics,
keyboard and mouse input, sound, and other common things required by games can take hours (trust me, I know).
With Velaptor, you can just focus on writing your game!!
```

Reviewer: I thought low-level was overused. Maybe:

Velaptor focuses on building your game instead of having to write low-level code. Rendering graphics, keyboard and mouse input, sound, and other common things required by games can take hours (trust me, I know).  With Velaptor, you can just focus on developing your game!!

Author: I like it.

:::info Quick Note

> Velaptor is currently still in preview and under development.  Though you can create games with Velaptor in
> its current state and it is always improving, further work still needs to be completed to get Velaptor to run on
> on other platforms such as Linux, Android, and iOS. 
> 
> This means that currently, Velaptor applications can only run on Windows.  
> As of right now, there are no plans to support MacOS.
> :::

Reviewer: Grammar
```suggestion
Velaptor is currently only in preview and under development.  You can create games with Velaptor in
its current state to run on Windows.  We are constantly adding improvements and there are future plans for Velaptor to run on other platforms such as Linux, Android, and iOS. As of right now, there are no plans to support MacOS.
:::
```

### Single Codebase

> The goal of Velaptor is to be able to write 2D games from a single code base and to run them on multiple platforms.
> This means you do not have to maintain multiple languages and tech stacks to be able to run your game on the desired platforms.

Reviewer: Grammar
```suggestion
The goal of Velaptor is to be able to write 2D games from a single code base and to run them on multiple platforms.
Now, you don't have to maintain multiple languages and tech stacks to be able to run your game on the desired platforms.
```

### Performant

> Velaptor has been built with performance in mind as much as possible and will always strive to keep
> things as performant as possible.

Reviewer: Grammar
```suggestion
Velaptor has been built with performance in mind as much as possible and will always strive to be as performant as possible.
```

Reviewer: You didn't make this change.

### Simple

> Creating games can be some of the most complicated and resource-demanding applications out there.  Our goal
> is to try and make creating games as simple and straight forward as possible.  This means we are willing to change
> the API if this means the gain in simplicity is substantial while keeping performance into account.
> 

Reviewer: Grammar
```suggestion
Creating games can be some of the most complicated and resource-demanding applications out there.  Our goal
is to make creating games as simple and straight forward as possible.  We are willing to change the API if it means there is a gain in simplicity, while keeping performance into account.

```

Reviewer: Sorry.  It should be: 

We are willing to change the API if it means there is a gain in simplicity, while taking performance into account.

### Reliable

We strive to make Velaptor as reliable and free of bugs as possible.  We all know that there is no such thing as
100% bug-free software.  But that does not mean we can't strive for it!!
Velaptor is highly tested.  We maintain a very high level of code coverage to make sure that nothing slips through

> the cracks and are always maintaining and improving our unit tests.

Reviewer: Grammar
```suggestion
We strive to make Velaptor as reliable and free of bugs as possible.  We all know that there is no such thing as
100% bug-free software but that does not mean we can't strive for it!!
Velaptor is highly tested.  We maintain a very high level of code coverage to make sure that nothing slips through
the cracks.  In addition, we are always maintaining and improving our unit tests.
```

> We have achieved this by designing the Velaptor architecture to be loosely coupled as well as making use of the observable

Reviewer: Grammar
```suggestion
We have achieved reliability by designing the Velaptor architecture to be loosely coupled as well as making use of the observable
```

This is one reason why we do lots of manual testing to cover the areas that unit tests cannot.
:::

## Join the community

We are always ready to welcome people to the community.  If you just want to say hi, ask some questions,
or ask how you can contribute to Velaptor or any of the many KinsonDigital organizational projects, we are here
to guide you.

- [Discord](https://discord.gg/qewu6fNgv7): For getting help, asking questions, conversation, and where we post updates and releases.
- [Twitter](https://twitter.com/KDCoder): Where we post updates on releases as well as other software related topics.
- [KinsonDigital](https://github.com/kinsondigital): The KinsonDigital organization where you can find Velaptor and as well as other projects.

## Learn how to contribute

The Velaptor project is run and maintained by [Calvin Wilkinson](https://twitter.com/KDCoder) and [Kristen Wilkinson](https://twitter.com/kswilky).  We work on this full-time
and work hard to make this project as well as all of the [KinsonDigital](https://github.com/KinsonDigital) projects in working order.  Any open-source developer will tell you
that maintaining open-source projects are a ton of work.🥱  

> Any contribution no matter how small is welcome and greatly appreciated.  If you are interested in contributing, refer to

Reviewer: Grammar
```suggestion
Any contribution, no matter how small, is welcome and greatly appreciated.  If you are interested in contributing, refer to
```

the links below for more info.
- Click [here](https://github.com/KinsonDigital/.github/blob/master/docs/CONTRIBUTING.md) to learn how to contribute your time.
- Click [here](https://github.com/sponsors/KinsonDigital) if you are interested in contributing your funds via one-time or recurring donations.

## License

> The Ionic UI Toolkit is a free and open-source project, released under the permissible [MIT license](https://github.com/KinsonDigital/Velaptor/blob/release/v1.0.0/LICENSE.md). This means it can be used in personal or commercial projects for free.

Reviewer: Grammar
```suggestion
The Ionic UI Toolkit is a free and open-source project, released under the permissible [MIT license](https://github.com/KinsonDigital/Velaptor/blob/release/v1.0.0/LICENSE.md) which means it can be used in personal or commercial projects for free.
```

## Overview

Velaptor focuses on building your game instead of having to write low-level code. Rendering graphics, keyboard and mouse input, sound, and other common things
required by games can take hours (trust me, I know). With Velaptor, you can just focus on developing your game!!

Writing these things from scratch can take hours when all you want to do is create.  This is where Velaptor can help!!

:::info Quick Note
Velaptor is currently only in preview and under development.  You can create games with Velaptor in
its current state to run on Windows.  We are constantly adding improvements and there are plans for Velaptor to run on other platforms such as
Linux, Android, and iOS. As of right now, there are no plans to support MacOS.
:::

### Single Codebase

The goal of Velaptor is to be able to write 2D games from a single code base and to run them on multiple platforms.
Now, you don't have to maintain multiple languages and tech stacks to be able to run your game on the desired platforms.

### Performant

> Velaptor has been built with performance in mind as much as possible and we will always strive to be as performant as possible.

Reviewer: Grammar
```suggestion
Velaptor has been built with performance in mind and we strive for it to be as performant as possible.
```

### Simple

Creating games can be some of the most complicated and resource-demanding applications out there.  Our goal is to try and make

> the process as simple as possible.  We are willing to change the API if it means there is a gain in simplicity while taking performance into account.

Reviewer: Grammar
```suggestion
Creating games can be some of the most complicated and resource-demanding applications out there.  Our goal is to make the process as simple as possible.  We are willing to change the API if it means there is a gain in simplicity while taking performance into account.
```
